client/signup/main.jsx

```diff
@@ -256,7 +256,12 @@ class Signup extends React.Component {
 
 		const {
 			initialContext: {
-				query: { vertical, site_type: siteType },
+				query: {
+					vertical,
+					site_type: siteType,
+					exclude_plan: excludePlanStep,
+					exclude_domain: excludeDomainStep,
```

Comment:
Are these coming from the query string? Would it make more sense to handle this at the framework level, and have mechanism in place in to skip steps in certain conditions? We could add some logic to the `domains-launch` and `plans-launch` steps which specifies the conditions under which they should be skipped.

Reply:
I've made a change in [2496746](https://github.com/Automattic/wp-calypso/pull/30242/commits/24967469a52b48ab9506c66038e1d111aaba5af0) to get the domains and plans for a site from the state tree itself. This works, but I'm not yet fully confident if the prop values `siteDomains` and `isPaidPlan` are always available in the component. I think it's in the right direction but needs a bit more testing.

Reply:
@scruffian Tested this for a while, and there is some inconsistent behaviour. 
On a free site with a registered domain, when I click the Launch Site button a few times, most of the times it correctly skips the domains step but there are a few times that it still shows the domain step. What I did:
1. Created a new private site with a plan + domain.
2. Cancelled the plan but kept the domain
3. Clicked the Launch Site button

The issue I see is that the `getState().sites.domains.items` is not always getting the domain values for the current site in the launch flow tab. It's somewhat random but reproducible. Would you be knowing why this is so?

Reply:
I wonder if the API isn't always getting called. Maybe you could add the QuerySiteDomains component to ensure that it is?

Reply:
@scruffian  I've pushed in a fix that uses the QuerySiteDomains component, but the issue still persists 😕 

Also, I’m not sure if skipping steps in the framework would always work. For example, we want to skip domain step in the launch flow but no in the Add Site flow. In both flows, the state tree has a `siteId`value so we would then have to add specific conditions such as skip if in launch flow but not otherwise.

Reply:
@scruffian This is now fixed in [ad07829](https://github.com/Automattic/wp-calypso/pull/30242/commits/ad078297780801cc3c03c0078eb69abfadbc73d6). The skipping of steps is now handled at the framework level.